README.md

```diff
@@ -1,23 +1,128 @@
-by the web client.
+## Example Usage
+
+This is a simplified example. See the data scripts in [`timelapse-colorizer-data`](./timelapse-colorizer-data/) for complete working examples!
+
+```python
+import pandas as pd
+from colorizer_data.writer import (
+    ColorizerDatasetWriter,
+    ColorizerMetadata,
+    FeatureInfo,
+    FeatureType,
+)
+from colorizer_data.utils import (
+    configureLogging,
+    generate_frame_paths,
+    remap_segmented_image,
+    make_bounding_box_array,
+)
+
+# Open a dataset
+data: pd.DataFrame = pd.read_csv("dataset.csv")
+output_dir = "./data"
+
+configureLogging(output_dir)
+writer = ColorizerDatasetWriter(output_dir, data)
+
+# Write data and features from the dataset
+writer.write_data(
+    tracks=data["tracks"],
+    times=data["times"],
+    outliers=data["outliers"],
+    centroids_x=data["centroids x"],
+    centroids_y=data["centroids y"]
+)
+writer.write_feature(
+    data["feature1"].to_numpy(),
+    FeatureInfo(label="My Feature", type=FeatureType.CONTINUOUS)
+)
+writer.write_feature(
+    data["feature2"].to_numpy(),
+    FeatureInfo(label="My Other Feature", type=FeatureType.CATEGORICAL, categories=["A", "B", "C"])
+)
+
+# Write frames and bounding boxes
+grouped_data = data.groupby("time")
+bounds_arr = make_bounding_box_array(grouped_data)
+for group_name, frame in grouped_data:
+    row = frame.iloc[0]
+    frame_number = row["time"]
+    # Get segmentations as a 2D array
+    seg2d: np.ndarray = row["2d-segmentation-image"]
+
+    # Remap the 2D array and write to the dataset as an image
+    seg_remapped, lut = remap_segmented_image(
+        seg2d,
+        frame,
+        "object-ids",
+    )
+    writer.write_image(seg_remapped, frame_number)
+    update_bounding_box_data(bounds_arr, seg_remapped)
+
+writer.write_data(bounds=bound_arr)
+writer.set_frame_paths(generate_frame_paths(len(grouped_data)))
+
+# Write manifest and finish
+metadata = ColorizerMetadata(
+    frame_width=400,
+    frame_height=300,
+    frame_units="µm"
+    frame_duration_sec=5,
+)
+writer.write_manifest(metadata=metadata)
+```
+
+## Versioning
+
+Colorizer-data uses [semantic versioning](https://semver.org). All versions are tagged in this repository with `vX.Y.Z`, where `X`, `Y`, and `Z` correspond with major, minor, and patch version numbers. The API will be backwards-compatible within the same major version.
+
+Contributing developers can update the version number using the [`bump-my-version`](https://github.com/callowayproject/bump-my-version) Python tool, which will automatically tag commits. You can learn more about [git tagging here.](https://git-scm.com/book/en/v2/Git-Basics-Tagging)
+
+### Basic Usage
+
+```
+pip install --upgrade bump-my-version
+
+# Do a dry run and check the output before updating!
+bump-my-version bump --dry-run -v [type]
+
+bump-my-version bump -v [type]
+git push origin [new tag]
```

Comment:
I think there might even be a simpler command like `git push origin --tags`.

Reply:
Interesting note on this! I saw on StackOverflow that developers are discouraged from running `git push origin --tags`, because it can also push potentially unwanted tags from other local development.

The two alternatives were pushing the specific tag name (current approach), or pushing with the `--folow-tags` option, which ignores unannotated or unreachable tags. I slightly prefer having devs push specific tag names as a validation step... what do you think?

Reply:
Makes sense.  I only make tags for versioning/releasing on main branch that I know should be in the origin remote.  So I would never have this issue.  But it's true you could be pushing local tags that would be meaningless and pollute the repo for others.  As long as you're aware of what it means.